lib/ansible/module_utils/basic.py

```diff
@@ -1583,32 +1569,15 @@ def _check_locale(self):
                            to_native(e), exception=traceback.format_exc())
 
     def _handle_aliases(self, spec=None, param=None):
-        # this uses exceptions as it happens before we can safely call fail_json
-        aliases_results = {}  # alias:canon
+        if spec is None:
+            spec = self.argument_spec
         if param is None:
             param = self.params
 
-        if spec is None:
-            spec = self.argument_spec
-        for (k, v) in spec.items():
-            self._legal_inputs.append(k)
-            aliases = v.get('aliases', None)
-            default = v.get('default', None)
-            required = v.get('required', False)
-            if default is not None and required:
-                # not alias specific but this is a good place to check this
-                raise Exception("internal error: required and default are mutually exclusive for %s" % k)
-            if aliases is None:
-                continue
-            if not isinstance(aliases, SEQUENCETYPE) or isinstance(aliases, (binary_type, text_type)):
-                raise Exception('internal error: aliases must be a list or tuple')
-            for alias in aliases:
-                self._legal_inputs.append(alias)
-                aliases_results[alias] = k
-                if alias in param:
-                    param[k] = param[alias]
-
-        return aliases_results
+        # this uses exceptions as it happens before we can safely call fail_json
+        alias_results, legal_inputs = handle_aliases(spec, param)
+        self._legal_inputs = legal_inputs
```

Comment:
Technically, you could remove this line, and just use:

```
alias_results, self._legal_inputs = handle_aliases(spec, param)
```

Reply:
Much nicer.